### PR TITLE
Assign ownership to many libs files

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Assign ownership to many of the remaining libs/common files.

Criteria for ownership:

- Files used by a single team, is now owned by that team.
- Files related to a domain owned by a team is now owned by that team.
- Where ownership is unclear the "lowest level" service takes ownership.

### Review notes

Please review the files moves first by searching for `→`. Flag early if you disagree with any of the moves related to your teams ownership.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by eliykat — DISMISSED** [review]

File: libs/common/src/auth/iframe-component.ts
Comment (jlf0dev) [inline libs/common/src/auth/iframe-component.ts:1]: Should this be platform? Feels like the captcha is auth specific but not the iframe itself

Comment (Hinton) [reply libs/common/src/auth/iframe-component.ts:1]: We only use iframes for auth concerns. Currently I can't think of any non auth related captcha can you?

**Review by MGibson1 — APPROVED** [review]
Great work, @Hinton. I know this was a slog, but so valuable.

I don't see any issues with the assignments. There are a few new files, but these appear to truly be new so no lost history either.

**Review by jlf0dev — DISMISSED** [review]

**Review by cagonzalezcs — APPROVED** [review]
Looks good from my side.

File: libs/common/src/services/search.service.ts
Comment (djsmith85) [inline libs/common/src/services/search.service.ts:1]: Who is going to own search?

File: libs/common/src/models/request/register.request.ts
Comment (djsmith85) [inline libs/common/src/models/request/register.request.ts:1]: Should the reqister be part of auth?

File: apps/web/src/app/settings/purge-vault.component.ts
Comment (djsmith85) [inline apps/web/src/app/settings/purge-vault.component.ts:1]: Should this be owned by vault?

File: apps/browser/src/background/webRequest.background.ts
Comment (djsmith85) [inline apps/browser/src/background/webRequest.background.ts:1]: This should be moved into ownership of @bitwarden/team-autofill-dev. This is related to autofilling/submitting basicAuth

File: apps/web/src/app/settings/delete-account.component.ts
Comment (djsmith85) [inline apps/web/src/app/settings/delete-account.component.ts:1]: This should be owned by owned by @bitwarden/team-auth-dev

File: apps/web/src/app/settings/api-key.component.ts
Comment (djsmith85) [inline apps/web/src/app/settings/api-key.component.ts:1]: The api-keys should likely be owned by @bitwarden/team-auth-dev

File: libs/common/src/models/request/register.request.ts
Comment (Hinton) [reply libs/common/src/models/request/register.request.ts:1]: It should, but moving requests are painful, and we're in the process of breaking out api services which would handle these ownerships.

**Review by djsmith85 — DISMISSED** [review]
Changes are looking good.
As a follow-up I have added some comments that @bitwarden/tech-leads should review and discuss. Ideally moving more files into ownership of teams.

@Hinton The only minor thing I had seen that would have been nice: If a file had a previous import from i.e. `vault/enums`, it would have been nice to combine them instead of having two.

File: libs/common/src/services/search.service.ts
Comment (Hinton) [reply libs/common/src/services/search.service.ts:1]: TBD. Realistically we might want to copy it and create a vault search and send search service. Or make a core search service and delegate details to models. But it doesn't have a clear owner yet.

File: apps/browser/src/background/models/addLoginQueueMessage.ts
Comment (djsmith85) [inline apps/browser/src/background/models/addLoginQueueMessage.ts:1]: Should be owned by @bitwarden/team-autofill-dev

**Review by cagonzalezcs — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

File: apps/browser/src/auth/background/service-factories/totp-service.factory.ts
Comment (djsmith85) [inline apps/browser/src/auth/background/service-factories/totp-service.factory.ts:1]: If the `totpService` is owned by vault, then the factory should also be owned by @bitwarden/team-vault-dev or is the assignment of the service wrong?

**Review by cagonzalezcs — DISMISSED** [review]

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/web/src/app/settings/change-kdf/change-kdf.component.ts
[the remark this answers is not in the document]
Comment (jlf0dev) [reply apps/web/src/app/settings/change-kdf/change-kdf.component.ts:1]: Most likely auth

**Review by jlf0dev — APPROVED** [review]
Pretty sure CLI build is failing for everyone

**Review by djsmith85 — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

File: apps/web/src/app/settings/change-kdf/change-kdf.component.ts
[the remark this answers is not in the document]
Comment (MGibson1) [reply apps/web/src/app/settings/change-kdf/change-kdf.component.ts:1]: yup, auth

**Review by cagonzalezcs — APPROVED** [review]

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Pull request merged**